dpcpp/CMakeLists.txt

```diff
@@ -57,6 +58,7 @@ target_compile_options(ginkgo_dpcpp PRIVATE "${GINKGO_DPCPP_FLAGS}")
 target_compile_features(ginkgo_dpcpp PRIVATE cxx_std_17)
 target_link_options(ginkgo_dpcpp PRIVATE -fsycl-device-lib=all)
 target_link_options(ginkgo_dpcpp PRIVATE -fsycl-device-code-split=per_kernel)
+target_link_libraries(ginkgo_dpcpp PRIVATE "mkl_sycl;mkl_intel_ilp64;mkl_sequential;mkl_core")
```

Comment:
Intel [provides one in their oneMKL library](https://github.com/oneapi-src/oneMKL/tree/develop/cmake), but _of course_ they don't ship it (or even better, a `MKLConfig.cmake`). In the past you [could get MKL](https://cmake.org/cmake/help/git-stage/module/FindBLAS.html) with
```
set(BLA_VENDOR Intel_*)
find_package(BLAS)
```
but they didn't update this setup for oneAPI yet.

Reply:
Great finding that they have one. I guess we can pay the price of including their license when using DPC++ in that case until something better is done about this.

Reply:
it does not link explicitly, so it will require compile flag public. Otherwise, it can not link to oneMKL in ginkgo_core
If you click link with oneMKL libraries explicitly, we do not need to make the compile flag public